crates/engine/tree/src/tree/precompile_cache.rs

```diff
@@ -0,0 +1,206 @@
+//! Contains a precompile cache that is backed by a moka cache.
+
+use alloy_primitives::Bytes;
+use reth_evm::precompiles::{DynPrecompile, Precompile};
+use revm::precompile::{PrecompileError, PrecompileResult};
+use std::sync::Arc;
+
+/// Cache for precompiles, for each input stores the result.
+#[derive(Debug, Clone)]
+pub struct PrecompileCache(
+    Arc<mini_moka::sync::Cache<CacheKey, CacheEntry, alloy_primitives::map::DefaultHashBuilder>>,
+);
+
+impl Default for PrecompileCache {
+    fn default() -> Self {
+        Self(Arc::new(
+            mini_moka::sync::CacheBuilder::new(100_000)
+                .build_with_hasher(alloy_primitives::map::DefaultHashBuilder::default()),
```

Comment:
let's create an issue to implement a custom `weigher` that measures the size in MB and limits by it, instead of constant 100k